### PR TITLE
support user defined attack

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
Close issue #169 

Users set the `attack-cmd` and `recover-cmd`, execute `attack-cmd` when attack, execute `recover-cmd` when recover.

```bash
chaosd attack user-defined -h
user defined attack related commands

Usage:
  chaosd attack user-defined <subcommand> [flags]

Flags:
  -a, --attack-cmd string    the command to be executed when attack
  -h, --help                 help for user-defined
  -r, --recover-cmd string   the command to be executed when recover

Global Flags:
      --log-level string   the log level of chaosd. The value can be 'debug', 'info', 'warn' and 'error'
      --uid string         the experiment ID
```

### CODE REVIEW

**Review by Andrewmatilde — APPROVED** [review]
LGTM

File: test/integration_test/user_defined/run.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+cur=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+cd $cur
+
+bin_path=../../../bin
+
+# test user defined attack
+${bin_path}/chaosd attack user-defined --attack-cmd "touch /tmp/chaos" --recover-cmd "rm /tmp/chaos" --uid 1234567890
+
+if [ ! -f /tmp/chaos ]; then
```
Comment (fewdan) [inline test/integration_test/user_defined/run.sh:26]: If other progress just created a directory called "/tmp/chaos", then "touch /tmp/chaos" will not report an error, but [ ! -f /tmp/chaos ] will get true, so is it possible to use '- e' would be more appropriate?

File: test/integration_test/user_defined/run.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Chaos Mesh Authors.
```
Comment (fewdan) [inline test/integration_test/user_defined/run.sh:3]: Change to "2022"?

**Review by fewdan — APPROVED** [review]
rest LGTM

**Review by cwen0 — APPROVED** [review]
LGTM

File: test/integration_test/user_defined/run.sh (hunk 1 above)
Comment (WangXiangUSTC) [reply test/integration_test/user_defined/run.sh]: done

**Pull request merged**